packages/react-reconciler/src/__tests__/ReactSuspenseyCommitPhase-test.js

```diff
@@ -279,4 +282,67 @@ describe('ReactSuspenseyCommitPhase', () => {
       </>,
     );
   });
+
+  // @gate enableOffscreen
+  test("host instances don't suspend during prerendering, but do suspend when they are revealed", async () => {
+    function More() {
+      Scheduler.log('More');
+      return <SuspenseyImage src="More" />;
+    }
+
+    function Details({showMore}) {
+      Scheduler.log('Details');
+      const more = useMemo(() => <More />, []);
+      return (
+        <>
+          <div>Main Content</div>
+          <Offscreen mode={showMore ? 'visible' : 'hidden'}>{more}</Offscreen>
+        </>
+      );
+    }
+
+    const root = ReactNoop.createRoot();
+    await act(async () => {
+      root.render(<Details showMore={false} />);
+      // First render the outer component, without the hidden content
+      await waitForPaint(['Details']);
+      expect(root).toMatchRenderedOutput(<div>Main Content</div>);
```

Comment:
shouldn't this be in `startTransition` too to demonstrate that its not just committing b/c of a sync render?

Reply:
The important part here is that the offscreen content is not being rendered, or even attempted. It's deferred until a later paint.

Reply:
but I suppose I could add another test to verify that it doesn't accidentally start becoming suspensey if the original update that spawned it was a transition.